### ISSUE TITLE
Shouldn't this library be under the GPL license?

### ISSUE THREAD

**Issue by arnavb** [peterbrittain/asciimatics#176]
This package relies on [pyfiglet](https://github.com/pwaller/pyfiglet), which is licensed under the GPLv2. 

Various sources say that importing a library under the GPL means that the importing code must also be GPL:
- https://opensource.stackexchange.com/a/6036
- https://stackoverflow.com/a/999543
- https://groups.google.com/d/msg/comp.lang.python/zYZGRRqs_Mk/S7vSwIy4ZzcJ

I don't even think the whole dynamic linking idea applies for the GPL (GNU doesn't even differentiate between static/dynamic linking):
- https://stackoverflow.com/a/40492802

The reason I worry about this is because I want to use this library under a MIT licensed application and not have to worry about switching to the GPL (I declare this as a dependency in a `setup.py`, if it matters).

If I am correct, is it possible you could either find an alternative to PyFiglet, or write an LGPL wrapper or something for it?

<sub>Yeah, I don't really like legal issues either, I just want to code.</sub>

**Comment by peterbrittain** [comment]
And now, I've raised an issue on the pyfiglet project to see if they will consider a weaker license...  Let's see what they say.

**Comment by arnavb** [comment]
@peterbrittain Ah, but the problem is, I _do_ need the figlet portion of the library!

I believe the options currently are:
1.  just asking PyFiglet to use a weaker license (although I doubt they will accept considering the point of the GPL).
2. Find an alternative to PyFiglet.

However, there is another (last-option) alternative that can be used if the previous options do not work. Since PyFiglet provides a [command line script](https://github.com/pwaller/pyfiglet/blob/master/setup.py#L54) for using it, you can decide to ditch the library interface and instead use it as an external program (maybe through `Popen` or similar). In this way, you are using just ["mere aggregation"](https://www.gnu.org/licenses/old-licenses/gpl-2.0-faq.en.html#MereAggregation) and don't have to change the license of this library to GPLv2+.

**Comment by peterbrittain** [comment]
Yeah - I saw that potential escape route too.  It's not very nice, so I'd rather not do that if I don't have to.  Let's see what pwaller says about re-licensing pyfiglet first...  If you're in a a hurry, I'd accept a PR to make this change.

**Comment by arnavb** [comment]
No, I'm not in any particular hurry as of now. I'm just going to avoid publishing my program to Github as of now, for the aforementioned reasons (its under the MIT license).

I might consider making a PR for the change you mentioned, but don't expect too much; I don't have that much time! 😓

**Comment by peterbrittain** [comment]
Yeah - we're all low on time.  Fingers crossed that things go well with https://github.com/pwaller/pyfiglet/issues/58

**Comment by peterbrittain** [comment]
Nearly there!  just 2 more developers to agree with the proposed license change!

**Comment by peterbrittain** [comment]
Unfortunately these 2 developers are not responding, so I've created a patch to use CLI invocation (with the option to fall back to the native figlet if present) instead.  See https://github.com/peterbrittain/asciimatics/commit/fdf63f0403e139e958e5a26cd700ba3c47f79ad1

**Comment by peterbrittain** [comment]
OK...  The use-figlet branch is now clean on Windows and Linux.  However, the run-time can be huge, depending on the start-up time for your figlet application, so I'm going to hold off merging until I really have to do so.

**Comment by arnavb** [comment]
Wow that took a while, but it could've been worse. Congrats!
